core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -20,27 +20,30 @@
 package org.apache.iceberg.hadoop;
 
 import com.google.common.base.Preconditions;
-import org.apache.iceberg.*;
-import org.apache.iceberg.catalog.TableIdentifier;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
-
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.exceptions.AlreadyExistsException;
 import org.apache.iceberg.exceptions.RuntimeIOException;
 
-import java.io.Closeable;
-import java.io.IOException;
-import java.util.Map;
 
 public class HadoopCatalog extends BaseMetastoreCatalog implements Closeable {
-  private final String ICEBERG_HADOOP_WAREHOUSE_BASE = "iceberg/warehouse";
-  private final org.apache.hadoop.conf.Configuration conf;
+  private static final String ICEBERG_HADOOP_WAREHOUSE_BASE = "iceberg/warehouse";
+  private final Configuration conf;
   private String hdfsRoot;
 
   public HadoopCatalog(Configuration conf) {
     this.conf = conf;
-    String hdfsRoot = conf.get("fs.defaultFS");
+    hdfsRoot = conf.get("fs.defaultFS");
```

Comment:
We should not use the `defaultFS` and allow the user to specify the right path [e.g they can pass the fs protocol as part of their path]